src/shared/parameter_v2/examples/generated_dynamic_parameters.hpp

```diff
@@ -0,0 +1,351 @@
+            std::const_pointer_cast<const Parameter<bool>>(foo_bool),
+            std::const_pointer_cast<const NumericParameter<int>>(foo_int),
+        };
+    }
+    
+    const std::shared_ptr<const Parameter<bool>> fooBool() const
+    {
+        return std::const_pointer_cast<const Parameter<bool>>(foo_bool);
+    }
+
+    const std::shared_ptr<Parameter<bool>> mutableFooBool()
+    {
+        return foo_bool;
+    }
+
+    const std::shared_ptr<const NumericParameter<int>> fooInt() const
+    {
+        return std::const_pointer_cast<const NumericParameter<int>>(foo_int);
+    }
+
+    const std::shared_ptr<NumericParameter<int>> mutableFooInt()
+    {
+        return foo_int;
+    }
+
+    const std::string name() const
+    {
+        return "foo"
+    }
+
+    bool loadFromCommandLineArguments(int argc, char **argv) {
+        struct commandLineArgs {
+            bool help = false;
+            bool foo_bool = true;
+            int foo_int = 3;
+        }
+
+        commandLineArgs args;
+        boost::program_options::options_description desc{"Options"};
+
+        desc.add_options()("help,h", boost::program_options::bool_switch(&args.help),
+                        "help screen");
+
+        desc.add_options()("foo_bool", boost::program_options::value<bool>(&args.foo_bool), "foo bool");
+        desc.add_options()("foo_int", boost::program_options::value<int>(&args.foo_int), "foo int");
+        
+        boost::program_options::variables_map vm;
+        boost::program_options::store(parse_command_line(argc, argv, desc), vm);
+        boost::program_options::notify(vm);
+
+        this->mutableFooBool()->setValue(args.foo_bool);
+        this->mutableFooInt()->setValue(args.foo_int);
+
+        if (args.help)
+        {
+            std::cout << desc << std::endl;
+        }
+
+        return args.help;
+    }
+
+    const MutableParameterList& getMutableParameterList()
+    {
+        return mutable_internal_param_list;
+    }
+
+    const ParameterList& getParameterList() const
+    {
+        return immutable_internal_param_list;
+    }
+
+   private:
+    MutableParameterList mutable_internal_param_list;
+    ParameterList immutable_internal_param_list;
+    std::shared_ptr<Parameter<bool>> foo_bool;
+    std::shared_ptr<NumericParameter<int>> foo_int;
+}
+
+class example : public Config
+{
+   public:
+    example(std::shared_ptr<foo> foo_config) : foo_config(foo_config)
+    {
+        example_bool_param = std::make_shared<Parameter<bool>>("example_bool_param", true);
+        example_int_param = std::make_shared<NumericParameter<int>>("example_int_param", 3, 0, 5);
+        example_unsigned_int_param = std::make_shared<NumericParameter<uint>>("example_unsigned_int_param", 3, 0, 5);
+        example_float_param = std::make_shared<NumericParameter<float>>("example_float_param", 4.04, 1.1, 9.01);
+        example_string_param = std::make_shared<Parameter<std::string>>("example_string_param", "Hello World");
```

Comment:
we need to also add the `example_factory_param` to see how that looks

Reply:
Done